llama.cpp

```diff
@@ -6420,10 +6420,13 @@ struct llama_grammar_candidate {
 // pointer. If an invalid sequence is encountered, returns `llama_partial_utf8.n_remain == -1`.
 static std::pair<std::vector<uint32_t>, llama_partial_utf8> decode_utf8(
         const char         * src,
+        size_t               n_src,
```

Comment:
Since this is a static function and seems to always be used with a std::string, why not remove the overload and change this one to use std::string?

Reply:
Didn't want to change the definition too much - was unsure if the author was considering exposing it via the C API at some point.

Reply:
I think it's fine. We'll probably need to work on the grammar further anyway. Will make a new issue to keep track.